### ISSUE TITLE
Issue in Math::GMPz::overload_pow()

### ISSUE THREAD

**Issue by trizen** [sisyphus/math-gmpz#3]
Hi,

There seems to be a minor issue in Math::GMPz::overload_pow() when the power is a Math::GMPz object and the base is a Perl integer.

The following code illustrates the problem:
```perl
use 5.014;
use warnings;

use Math::GMPz;

my $p = Math::GMPz->new(5);

say $p**3;    # this works
say 3**$p;    # this fails
```

Output:
```text
125
Invalid third argument (&PL_sv_yes) supplied to Math::GMPz::overload_pow function at z.pl line 10.
```

**Comment by sisyphus** [comment]
I think (but need to double check) that this is already fixed in github version.
I also need to check that there's a test that catches this case.
I'll get my act together and release 0.45 over this weekend (hopefully tonight).

Cheers,
Rob

**Comment by sisyphus** [comment]
Hmmm ... for a perl integer $i and Math::GMPz, object $p, I'm only allowing $i ** $p to work if $p holds a value that fits into an 'unsigned long int'. Otherwise the operation is a fatal error.
I do this because gmp itself allows only for an exponent value that fits into an 'unsigned long int'.

However, Math::GMP seems to allow $p to hold a value that is too large to fit into an unsigned long int.
What do you think ?

I'm thinking that the current github version is ready to be released as 0.45.
All the better if you get a chance to give it a try before I release it tomorrow night.

Thank you for your continuing feedback - it's much appreciated.

Cheers,
Rob

**Comment by trizen** [comment]
I can confirm the fix in the Github version.

Regarding the check for the value of $p, which must fit into an unsigned long int, you did a good choice checking it before it overflows with a core dump, as in Math::GMP, although I don't think that ever happens in practice.

Thank you very much for your time and passion.

All the best,
Daniel
